dev/com.ibm.ws.cdi.interfaces/src/com/ibm/ws/cdi/CDIService.java

```diff
@@ -68,4 +72,12 @@ public interface CDIService {
      */
     public boolean isWeldProxy(Object obj);
 
+    /**
+     * Returns all interceptor bindings on the invocation context.
+     *
+     * @return a set of interceptor bindings from the invocation context. This will include all interceptors that will run, are running, and have already run.
+     * @throws IllegalArgumentException if InvocationContext is not an instance of org.jboss.weld.interceptor.proxy.AbstractInvocationContext;
+     */
+    public Set<Annotation> getInterceptorBindingsFromInvocationContext(InvocationContext ic) throws IllegalArgumentException;
```

Comment:
Does it include interceptor bindings that didn't actually cause any interceptors to be bound? Probably need to add this to the test.

If so, you could say:
```suggestion
    /**
     * Returns all interceptor bindings which apply to the current invocation or lifecycle event.
     *
     * @return a set of interceptor bindings which apply to the current invocation or lifecycle event. This will include all interceptor bindings that apply, not just those that were used to bind the current interceptor.
     * @throws IllegalArgumentException if InvocationContext is not an instance of org.jboss.weld.interceptor.proxy.AbstractInvocationContext;
     */
    public Set<Annotation> getInterceptorBindingsFromInvocationContext(InvocationContext ic) throws IllegalArgumentException;
```